src/index.js

```diff
@@ -1447,5 +1452,266 @@ module.exports = {
+        score++;
+      }
+
+      if (typeof element.hasAttribute === 'function') {
+        const attributes = spec.attributes || {};
+        const className = attributes['class'];
+        const style = attributes.style;
+
+        if (className && element.hasAttribute('class')) {
+          if (typeof className === 'string') {
+            const expectedClasses = getClassNamesFromAttributeValue(className);
+            const actualClasses = getClassNamesFromAttributeValue(
+              element.getAttribute('class')
+            );
+
+            expectedClasses.forEach(expectedClass => {
+              if (actualClasses.indexOf(expectedClass) !== -1) {
+                score++;
+              }
+            });
+          } else if (isTextMatching(element.getAttribute('class'), className)) {
+            score++;
+          }
+        }
+
+        if (style && element.hasAttribute('style')) {
+          const expectedStyles =
+            typeof style === 'string' ? styleStringToObject(style) : style;
+          const actualStyles = styleStringToObject(
+            element.getAttribute('style')
+          );
+
+          Object.keys(expectedStyles).forEach(styleName => {
+            const expectedStyle = expectedStyles[styleName];
+            const actualStyle = actualStyles[styleName];
+
+            if (actualStyle) {
+              score++;
+            }
+
+            if (isTextMatching(actualStyle, expectedStyle)) {
+              score++;
+            }
+          });
+        }
+
+        const specialAttributes = ['style', 'class'];
+        const ids = ['id', 'data-test-id', 'data-testid'];
+
+        Object.keys(attributes).forEach(attributeName => {
+          if (specialAttributes.indexOf(attributeName) !== -1) {
+            return; // skip
+          }
+
+          if (element.hasAttribute(attributeName)) {
+            if (typeof attributes[attributeName] === 'boolean') {
+              score++;
+            }
+
+            if (
+              element.getAttribute(attributeName) === attributes[attributeName]
+            ) {
+              score += ids.indexOf(attributeName) !== -1 ? 100 : 1;
+            }
+          } else if (!attributes[attributeName]) {
+            score++;
+          }
+        });
+      }
+
+      const expectedChildren = spec.children || [];
+
+      expectedChildren.forEach((childSpec, i) => {
+        const child = element.childNodes[i];
+        const childType = expect.findTypeOf(child);
+
+        if (typeof childSpec.nodeType === 'number') {
+          if (child && child.nodeType === childSpec.nodeType) {
+            if (childType.is('DOMElement')) {
+              // Element
+              score += scoreElementAgainstSpec(
+                element.childNodes[i],
+                convertDOMNodeToSatisfySpec(childSpec)
+              );
+            }
+
+            score++;
+          } else if (expect.findTypeOf(childSpec).is('DOMIgnoreComment')) {
+            score++;
```

Comment:
Shouldn't this be neutral score-wise? Also, it looks like we'll end up here even if we're past `element.childNodes.length`?

Reply:
We are looking for similarities so you get a point for matching ignore.

 I'll take a closer look at the code around the length, as it is a bit hard on the phone 😊

Reply:
Thanks for pointing out the problem around the number of children. That was actually an error. I fixed that and added a good amount of tests around it.